commatrix/commatrix_test.go

```diff
@@ -50,3 +61,72 @@ func TestWriteMatrixToFile(t *testing.T) {
 	assert.NoError(t, err)
 	assert.FileExists(t, filepath.Join(destDir, "test-matrix.json"))
 }
+
+func TestGenerateSS(t *testing.T) {
+	clientset := fake.NewSimpleClientset()
+
+	ctrlTest := gomock.NewController(t)
+	defer ctrlTest.Finish()
+
+	mockDebugPod := debug.NewMockDebugPodInterface(ctrlTest)
+
+	tcpOutput := []byte(`LISTEN 0      4096      127.0.0.1:8797  0.0.0.0:* users:(("machine-config-",pid=3534,fd=3))                
+LISTEN 0      4096      127.0.0.1:8798  0.0.0.0:* users:(("machine-config-",pid=3534,fd=13))               
+LISTEN 0      4096      127.0.0.1:9100  0.0.0.0:* users:(("node_exporter",pid=4147,fd=3))`)
+
+	udpOutput := []byte(`UNCONN 0      0           0.0.0.0:111   0.0.0.0:* users:(("rpcbind",pid=1399,fd=5),("systemd",pid=1,fd=78))
+UNCONN 0      0         127.0.0.1:323   0.0.0.0:* users:(("chronyd",pid=1015,fd=5))                        
+UNCONN 0      0      10.46.97.104:500   0.0.0.0:* users:(("pluto",pid=2115,fd=21))`)
+	Output := []byte(`1: /system.slice/containerd.service
+	2: /system.slice/kubelet.service
+	3: /system.slice/sshd.service`)
+
+	// Set up the expectations
+	mockDebugPod.EXPECT().ExecWithRetry(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(
+		func(cmd string, interval, duration time.Duration) ([]byte, error) {
+			if strings.HasPrefix(cmd, "cat /proc/") && strings.Contains(cmd, "/cgroup") {
+				return Output, nil
+			}
+			if cmd == "ss -anpltH" {
+				return tcpOutput, nil
+			}
+			if cmd == "ss -anpluH" {
+				return udpOutput, nil
+			}
+			return nil, fmt.Errorf("unknown command")
+		},
+	).AnyTimes()
+
+	mockDebugPod.EXPECT().Clean().Return(nil).AnyTimes()
+
+	mockDebugPod.EXPECT().GetNodeName().Return("test-node").AnyTimes()
+
+	originalNew := debug.NewDebugPod
+	debug.NewDebugPod = func(cs *clientutil.ClientSet, node string, namespace string, image string) (debug.DebugPodInterface, error) {
+		return mockDebugPod, nil
+	}
+	defer func() { debug.NewDebugPod = originalNew }()
```

Comment:
Another way of doing it, is we could just create a func `func NewFakeDebugPod() DebugPod` within this file.

Since we passing around interfaces now, you shouldn't have to spoof `New()` at all.  Just setup your `DebugPod` struct with everything it needs before passing it along to the funcs that accept a `DebugPodInterface`.

Reply:
This would leave `New()` untouched and you wouldn't need `var NewDebugPod = New` in the `debug/debug.go` file.